### ISSUE TITLE
Headers focus and keyboard navigation for Pivot Grid Row Dimension Headers. 

### ISSUE THREAD

**Issue by skrustev** [IgniteUI/igniteui-angular#14249]
## Is your feature request related to a problem? Please describe.
Add focus/activation for igxPivotGrid row dimension headers on click. 

## Describe the solution you'd like
Using keyboard navigation the user should be able to go from row headers to row dimension header to column headers and back. Pretty much go to any pivot grid headers when the row dimension headers are enabled.

Sorting should be possible throw keyboard interaction as well.

Currently dimension expand/collapse through keyboard interaction is not possible, so maybe include that as well?
